tests/base_scraper/test_base_scraper.py

```diff
@@ -5,133 +5,120 @@
 import os
-import unittest
-
-import mozhttpd_base_test as mhttpd
+import pytest
 import requests
 
 import mozdownload
 import mozdownload.errors as errors
 from mozdownload.scraper import PLATFORM_FRAGMENTS
 from mozdownload.utils import create_md5, urljoin
 
-
-class TestBaseScraper(mhttpd.MozHttpdBaseTest):
-    """Testing the basic functionality of the Base Scraper Class"""
-
-    def test_platform_regex(self):
-        """Test for correct platform_regex output"""
-
-        for key in PLATFORM_FRAGMENTS:
-            scraper = mozdownload.Scraper(destination=self.temp_dir,
-                                          platform=key)
-            self.assertEqual(scraper.platform_regex,
-                             PLATFORM_FRAGMENTS[key])
-
-    def test_download(self):
-        """Test download method"""
-
-        filename = 'download_test.txt'
-        # standard download
-        test_url = urljoin(self.wdir, filename)
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=self.temp_dir,
-                                            logger=self.logger)
+@pytest.mark.parametrize('platform_key,platform_value', PLATFORM_FRAGMENTS.items())
+def test_platform_regex(tmpdir, platform_key, platform_value):
+    """Test for correct platform_regex output"""
+    scraper = mozdownload.Scraper(destination=str(tmpdir), platform=platform_key)
+    assert scraper.platform_regex == platform_value
+
+def test_standard_download(httpd, tmpdir):
+    """Test standard download method"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=str(tmpdir))
+    scraper.download()
+    assert os.path.isfile(os.path.join(str(tmpdir), filename))
+
+def test_compare_download(httpd, tmpdir):
+    """Compare original and downloaded file via md5 hash"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=str(tmpdir))
+    scraper.download()
+    md5_original = create_md5(os.path.join(httpd.router.doc_root, filename))
+    md5_downloaded = create_md5(os.path.join(str(tmpdir), filename))
+    assert md5_original == md5_downloaded
+
+def test_request_exception(httpd, tmpdir):
+    """RequestException"""
+    test_url1 = urljoin(httpd.get_url(), 'does_not_exist.html')
+    scraper1 = mozdownload.DirectScraper(url=test_url1, destination=str(tmpdir))
+    with pytest.raises(requests.exceptions.RequestException):
+        scraper1.download()
+
+def test_retry_attempts(httpd, tmpdir):
+    """Covering retry attempts"""
+    test_url2 = urljoin(httpd.get_url(), 'does_not_exist.html')
+    scraper2 = mozdownload.DirectScraper(url=test_url2,
+                                         destination=str(tmpdir),
+                                         retry_attempts=3,
+                                         retry_delay=1.0)
```

Comment:
Please remove or reduce the delay to something around 100ms. It won't make a difference anyway.